iocage_lib/ioc_create.py

```diff
@@ -547,26 +549,44 @@ def create_install_packages(self, jail_uuid, location, config,
             if r and len(r.groups()) >= 3:
                 repo = r.group(3)
 
+            iocage_lib.ioc_common.logit({
+                "level": "INFO",
+                "message": f"\nTesting Host DNS response to {repo}"
+            },
+                _callback=self.callback,
+                silent=False)
+
+            try:
+                dns.resolver.query(repo)
+            except dns.resolver.NoNameservers:
+                iocage_lib.ioc_common.logit({
+                    "level": "EXCEPTION",
+                    "message": f"{repo} could not be reached via DNS, check"
+                    " your network"
+                },
+                    _callback=self.callback,
+                    silent=False)
+
         # Connectivity test courtesy David Cottlehuber off Google Group
-        # TODO: Use python's dns.resolver here?
         srv_connect_cmd = ["drill", "-t", f"_http._tcp.{repo} SRV"]
         dnssec_connect_cmd = ["drill", "-D", f"{repo}"]
         dns_connect_cmd = ["drill", f"{repo}"]
 
         iocage_lib.ioc_common.logit({
             "level": "INFO",
-            "message": f"\nTesting SRV response to {site}"
+            "message": f"Testing {jail_uuid}'s SRV response to {repo}"
         },
             _callback=self.callback,
             silent=False)
 
-        # msg_err_return is set to silence stderr, we tell them user that
-        # error before
-        _, _, srv_err = iocage_lib.ioc_exec.IOCExec(
-            srv_connect_cmd, jail_uuid, location, plugin=self.plugin,
-            silent=True, msg_err_return=True).exec_jail()
+        try:
+            with iocage_lib.ioc_exec.IOCExec(
+                srv_connect_cmd, jail_uuid, location, plugin=self.plugin
+            ) as _exec:
+                out = _exec.exec_jail()
+                iocage_lib.ioc_common.consume_and_log(out, silent=True)
```

Comment:
Why do you still have to consume it? Do you think there is a way to work it out so its consumed automatically in the context manager `__exit__`?